src/driver/amdxdna/amdxdna_ctx.c

```diff
@@ -972,92 +964,71 @@ int amdxdna_drm_exec_cmd_ioctl(struct drm_device *dev, void *data, struct drm_fi
 		goto free_bo_hdls;
 	}
 
-	/*
-	 * SRCU lock for synchronizing with amdxdna_hwctx_destroy_rcu().
-	 * I don't worry about concurrently stop/restart context. Because this
-	 * ioctl just create job and push it to DRM's queue. When stop/restart
-	 * context, DRM scheduler has protection for the jobs in its queue.
-	 *
-	 * Just make sure before this ioctl exited, don't release context.
-	 */
-	idx = srcu_read_lock(&client->hwctx_srcu);
-
-	gobj = drm_gem_object_lookup(filp, cmd_bo);
-	if (!gobj) {
-		XDNA_ERR(xdna, "Lookup GEM object failed");
+	cmd_bo = amdxdna_gem_get_obj(dev, cmd_bo_hdl, AMDXDNA_BO_CMD, filp);
+	if (!cmd_bo) {
+		XDNA_DBG(xdna, "get cmd bo failed");
 		ret = -ENOENT;
-		goto unlock_hwctx_srcu;
-	}
-
-	type = amdxdna_gem_get_obj_type(gobj);
-	if (unlikely(type != AMDXDNA_GEM_OBJ)) {
-		XDNA_ERR(xdna, "Invalid exec cmd BO type %d", type);
-		ret = -EINVAL;
-		goto release_cmd_obj;
+		goto free_bo_hdls;
 	}
-	abo = to_xdna_gem_obj(gobj);
 
-	if (abo->base.size < sizeof(struct amdxdna_cmd)) {
-		XDNA_ERR(xdna, "Bad cmd BO size: %ld", abo->base.size);
+	if(to_gobj(cmd_bo)->size < sizeof(struct amdxdna_cmd)) {
```

Comment:
No `clang-tidy` or `clang-format` or whatever?

Reply:
CI runs internally. Will add clang-tidy, probably in pre-commit hook.